integration/test_integration_basics.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
```

Comment:
Might be best to go straight to python3, as Envoy is mid-way through the slow march to Python 3 (Google import will also require this).

Reply:
I have had a fight with bazel/CI on this topic. I agree, and would love to move to Python 3, but are you OK with postponing that to a next PR? 
(I think we need another Envoy dependency upgrade / upgrade the CI image)

Reply:
I'll add an item to track this and attempt to move to Python3 in a follow up where we update the ci image and our Envoy dependency. 

With Python3 I think we would also have a fairly easy path to enhancing the integration tests with parallel execution 
& parameterized tests support.